lib/utils/getAPI.dart

```diff
@@ -1,29 +1,107 @@
-/*import 'package:http/http.dart' as http;
+import 'dart:io';
+
+import 'package:http/http.dart' as http;
 import 'dart:convert';
 
-class CardsData {
-  static Future<String> getJson(String url, String outgoing) async{
-    String ret = "";
+import 'package:smart_chef/utils/globals.dart';
+
+class UserData {
+  static Future<http.Response> authUser(String url, String outgoing) async{
+    http.Response response;
+    final headers = {HttpHeaders.contentTypeHeader: 'application/json'};
     try {
-      http.Response response = await http.post(url,
+      String totalurl = '$API_PREFIX$url';
+      response = await http.post(Uri.parse(totalurl),
           body: utf8.encode(outgoing),
-          headers:  {
-        "Accept":"application/json",
-            "Content-Type":"application/json",
-          },
+          headers:  headers,
       encoding:Encoding.getByName("utf-8"));
-      ret = response.body;
+      return response;
+    }
+    catch(e) {
+      print("Could not connect!");
+    }
+    throw Exception('Could not connect to server');
+  }
+
+  static Future<http.Response> getUser(String url, String auth) async{
+    http.Response response;
+    final headers = {HttpHeaders.contentTypeHeader: 'application/json', HttpHeaders.authorizationHeader: auth};
+    try {
+      String totalurl = '$API_PREFIX$url';
+      response = await http.get(Uri.parse(totalurl), headers: headers);
+      return response;
+    }
+    catch(e) {
+      print(e.toString());
+    }
+    throw Exception('Could not connect to server');
+  }
+
+  static Future<http.Response> logoutUser(String url, String auth) async{
+    http.Response response;
+    final headers = {HttpHeaders.contentTypeHeader: 'application/json', HttpHeaders.authorizationHeader: auth};
+    try {
+      String totalurl = '$API_PREFIX$url';
+      response = await http.get(Uri.parse(totalurl), headers: headers);
+      return response;
+    }
+    catch(e) {
+      print(e.toString());
+    }
+    throw Exception('Could not connect to server');
```

Comment:
This is identical for most of your requests; could you extract one of these for each of GET, PUT, POST, and DELETE?